py_neuromodulation/nm_stream.py

```diff
@@ -117,42 +129,58 @@ def _run(
 
             generator = self.lsl_stream.get_next_batch()
 
-        l_features: list[dict] = []
-        last_time = None
+        prev_batch_end = 0
 
         while True:
             next_item = next(generator, None)
 
             if next_item is not None:
-                time_, data_batch = next_item
+                timestamps, data_batch = next_item
```

Comment:
I changed this from time_ to timestamps, although I'm not sure is consistent with RawGenerator at this point